src/Controls/StreetControls.js

```diff
@@ -77,8 +113,74 @@ class StreetControls extends FirstPersonControls {
         this.previousPosition = newPreviousPosition;
     }
 
+    pick(event, pickGround = () => {}, pickObject = () => {}, target = new THREE.Vector3()) {
+        // get real distance to ground, with a specific method to pick on the elevation layer
+        this.view.getPickingPositionFromDepth(this.view.eventToViewCoords(event), target);
+        const distanceToGround = this.camera.position.distanceTo(target);
+
+        const intersectsBuildings = this.view.pickObjectsAt(event, 0, this.buildingsLayer);
+        const intersectsGround = this.view.pickObjectsAt(event, 0, this.view.tileLayer);
```

Comment:
you could do the ground picking only if there's not picking with the building

Reply:
Yes